clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6907,14 +6907,18 @@ void OffloadBundler::ConstructJobMultipleOutputs(
   InputInfo Input = Inputs.front();
   const char *TypeArg = types::getTypeTempSuffix(Input.getType());
   const char *InputFileName = Input.getFilename();
+  bool IsMSVCEnv =
+      C.getDefaultToolChain().getTriple().isWindowsMSVCEnvironment();
 
-  // For objects, we have initial support for fat archives (archives which
+  // For Linux, we have initial support for fat archives (archives which
   // contain bundled objects). We will perform partial linking against the
-  // object and specific offload target archives which will be sent to the
-  // unbundler to produce a list of target objects.
-  if (!C.getDefaultToolChain().getTriple().isWindowsMSVCEnvironment() &&
-      Input.getType() == types::TY_Object &&
-      TCArgs.hasArg(options::OPT_foffload_static_lib_EQ)) {
+  // specific offload target archives which will be sent to the unbundler to
+  // produce a list of target objects.
+  // FIXME: This should be a separate job in the toolchain.
+  if (!IsMSVCEnv && (Input.getType() == types::TY_Archive ||
+       Input.getType() == types::TY_WholeArchive ||
+       Input.getType() == types::TY_Object) &&
```

Comment:
I'd suggest running `Input.getType()` once before the conditional block so as to lessen verbosity

Reply:
Done.